handlers/ocr-handler/server.py

```diff
@@ -115,12 +115,50 @@ def render_ocr():
         logging.debug("Sending response")
         return response
 
-    # Get text renderer data
-    text = 'The following ' + str(len(ocr_data['lines']))
-    text += ' lines were found in the image: '
-    for i, line in enumerate(ocr_data['lines']):
-        line_text = line['text'] + '\n'
-        text += line_text
+    # Text to be returned
+    text = ""
+
+    # Object detection data is present
+    od = 'ca.mcgill.a11y.image.preprocessor.objectDetection'
+    if od in preprocessors and len(preprocessors[od]['objects']) > 0:
+        object_data = preprocessors[od]
+        retmaining_text = ocr_data['lines']
+        remaining_objects = [{key: obj[key] for key
+                              in ['type', 'dimensions']} for
+                             obj in object_data['objects']]
+        text += "The following objects were detected: "
+        for obj in remaining_objects:
+            obj_dims = get_obj_dims(obj)
+            obj_text = ""
+            for i, line in enumerate(retmaining_text):
+                lines_to_remove = []
+                text_dims = get_text_dims(line)
+                if is_contained(text_dims, obj_dims):
+                    obj_text += line['text'] + ", "
+                    lines_to_remove.append(i)
+            # Add the appropraite article of the object
+            # as well as the object type to the text
+            text += get_article(obj['type']) + obj['type']
+            if len(obj_text) > 0:
+                obj_text = obj_text[:-2]
+                text += "containing the text: " + obj_text + ". "
```

Comment:
I'm a little uncertain about "containing" as the operative word here...I think it may be clear for something like a sign where its relationship to text is obvious, but this may be weird for other objects. Saying "a bus containing the text: STM" or "a person containing the text..." strikes me as weird. I don't have a good alternative though other than making it more explicit that it's the region of the image detected as x also contains some text y. I think this can probably stay as is for now but pinging @Cybernide in case I'm missing some obvious alternative